### PR TITLE
[12.0][FIX] mass_mailing_list_dynamic: Prevent ValidationError when merge contacts

### PR DESCRIPTION
**Pull request by victoralmau**
Prevent ValidationError when merge contacts wizard related to dynamic and full sync list.
Similar to https://github.com/OCA/social/pull/633

Need to be fixed in 13.0 too.

Please @pedrobaeza and @chienandalu  can you review it?

@Tecnativa TT28777

### CODE REVIEW

**Review by chienandalu — APPROVED** [review]

File: mass_mailing_list_dynamic/wizards/partner_merge.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+from odoo import models
+
+
+class BasePartnerMergeAutomaticWizard(models.TransientModel):
+    _inherit = "base.partner.merge.automatic.wizard"
+
+    def _merge(self, partner_ids, dst_partner=None, extra_checks=True):
+        return super(
+            BasePartnerMergeAutomaticWizard, self.with_context(
+                partner_merge_wizard=True
```
Comment (pedrobaeza) [inline mass_mailing_list_dynamic/wizards/partner_merge.py]: Isn't there already any context to use?

**Review by pedrobaeza — COMMENTED** [review]
Why not avoid the exception condition in advance? Or is not possible? If not, put a generic context key like `bypass_dynamic_list_check` for being used by any other dependent module.

File: mass_mailing_list_dynamic/wizards/partner_merge.py (hunk 1 above)
Comment (victoralmau) [reply mass_mailing_list_dynamic/wizards/partner_merge.py]: Not exists any value in context to re-use it here, changed to `bypass_dynamic_list_check`.

File: mass_mailing_list_dynamic/models/mail_mass_mailing_contact.py [hunk 2]
```diff
@@ -11,7 +11,10 @@ class MassMailingContact(models.Model):
     _inherit = "mail.mass_mailing.contact"
 
     def _check_dynamic_full_sync_list(self, mailing_list):
-        if mailing_list.dynamic and mailing_list.sync_method == "full":
+        if (
+            mailing_list.dynamic and mailing_list.sync_method == "full"
+            and "bypass_dynamic_list_check" not in self.env.context
```
Comment (pedrobaeza) [inline mass_mailing_list_dynamic/models/mail_mass_mailing_contact.py]: You should explicitly check the value of the context

```suggestion
            and self.env.context.get("bypass_dynamic_list_check")
```

**Review by pedrobaeza — APPROVED** [review]

**Comment by pedrobaeza** [comment]
/ocabot merge patch

I suppose this is needed in v13 as well

**Pull request merged**